pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -590,6 +602,55 @@ func validateDomain(domain string, validDomains []string) bool {
 	return false
 }
 
+func validateMachinePools(newObject *hivev1.ClusterDeployment, contextLogger *log.Entry) *admissionv1beta1.AdmissionResponse {
+
+	if reason, result := validateMachinePool(newObject, &newObject.Spec.ControlPlane, contextLogger); !result {
+		message := fmt.Sprintf("Control plane machine pool validation failure: %s", reason)
+		return &admissionv1beta1.AdmissionResponse{
+			Allowed: false,
+			Result: &metav1.Status{
+				Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+				Message: message,
+			},
+		}
+	}
+
+	for _, mp := range newObject.Spec.Compute {
```

Comment:
Don't we also need to validate that there exists a "worker" compute pool? Otherwise, the defaults will be applied for that pool by the installer.

Reply:
this is an interesting question. i suppose there just would be nothing to sync if you provided an empty worker pool.

Reply:
Fair point.